### PR TITLE
Add mcount metrics

### PR DESCRIPTION
**Pull request by sowjumn**
Add metrics for different golang version usage.

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -403,6 +412,8 @@ trap clearGitCredHelper INT TERM EXIT
 
 determineTool
 
+mcount "version.$ver"
```
Comment (jmorrell) [inline bin/compile]: There are other binaries we might track versions of, like `dep`, etc. How about making this:

```
mcount "version.go.$ver
```

File: bin/compile [hunk 2]
```diff
@@ -403,6 +412,8 @@ trap clearGitCredHelper INT TERM EXIT
...
+
 ver=$(expandVer $ver)
```
Comment (jmorrell) [inline bin/compile]: if I understand correctly, this may modify the `ver` variable. What if we added the `mcount` inside the `reportVer` function instead?

**Review by jmorrell — CHANGES_REQUESTED** [review]

File: bin/compile (hunk 1 above)
Comment (freeformz) [reply bin/compile]: isn't that handled by BPLOG_PREFIX?

Comment (freeformz) [reply bin/compile]: Oh nevermind. Now I understand. I'm slow. ;-)

File: bin/compile (hunk 2 above)
Comment (freeformz) [reply bin/compile]: We should record the initial version and the expanded version IMO.

File: bin/compile (hunk 1 above)
Comment (sowjumn) [reply bin/compile]: @freeformz @jmorrell these are excellent suggestions. What do you think of 
BPLOG_PREFIX="buildpack.golang" with mcount "version.go.$ver

File: bin/compile (hunk 2 above)
Comment (sowjumn) [reply bin/compile]: @jmorrell @freeformz expandVer modifies the `ver` variable with go${v} . I was thinking that as long as we capture the version we should be fine and hence I placed it there. But lets chat about this.

Comment (jmorrell) [reply bin/compile]: The issue is that you'll capture both: `1.12` and `go1.12` and count them as separate things when you place it here.

That info is also useful, but it's probably not what you want when capturing "which go version was installed"

File: bin/compile (hunk 1 above)
Comment (jmorrell) [reply bin/compile]: I lean towards `BPLOG_PREFIX="buildpack.go"` so that it matches the buildpack name, but either works.

**Review by jmorrell — APPROVED** [review]
Thanks for the changes 👍

**Pull request merged**